internal/jobobject/jobobject.go

```diff
@@ -503,6 +507,27 @@ func (job *JobObject) ApplyFileBinding(root, target string, merged bool) error {
 	return nil
 }
 
+// isOpenedJobSilo is a helper to determine if a job object that was opened is a silo. This should ONLY be called
+// from `Open` and any callers in this package afterwards should use `job.isSilo()``
+func (job *JobObject) isOpenedJobSilo() bool {
+	// None of the information from the structure that this info class expects will be used, this is just used as
+	// the call will fail if the job hasn't been upgraded to a silo so we can use this to tell when we open a job
+	// if it's a silo or not. Because none of the info matters simply define a dummy struct with the size that the call
+	// expects which is 16 bytes.
+	type isSiloObj struct {
+		_ [16]byte
+	}
+	var siloInfo isSiloObj
+	err := winapi.QueryInformationJobObject(
+		job.handle,
+		winapi.JobObjectSiloBasicInformation,
+		uintptr(unsafe.Pointer(&siloInfo)),
```

Comment:
As an aside, it's probably not safe to pass a `uintptr` to this function. We should pass a `unsafe.Pointer` and only convert to `uintptr` at the syscall site.

Reply:
We alternate between [using LocalAlloc](https://github.com/microsoft/hcsshim/blob/06ce0c3c367904ee6e27130344b021bb66007c7c/internal/safefile/safeopen.go#L175) and [converting to `*byte` buffers](https://github.com/helsaawy/hcsshim/blob/ce36677c16415736a1cb3cc51e80b6dc69dc4107/internal/uvm/vsmb.go#L155), so if we do a separate PR, we may want to audit for other potentially unsafe calls and standardize on one approach.

Reply:
Oddly enough go vet doesn't whine about this case (or the others)

Reply:
To close on this https://github.com/microsoft/hcsshim/pull/1438. We can get that PR checked in and I can rebase this

Reply:
Rebasing shortly to fix this

Reply:
Rebased, resolving!